src/ZoneServer/Commands/ChatCommands.Handlers.cs

```diff
@@ -327,6 +330,118 @@ private CommandResult HandleJump(Character sender, Character target, string mess
+			if (args.Count < 2)
+			{
+				Log.Debug("HandlePetHire: Invalid call by user '{0}': {1}", sender.Username, command);
+				return CommandResult.Okay;
+			}
+
+			if (sender.Companions.HasCompanions)
+				return CommandResult.Okay;
+
+			if (!int.TryParse(args.Get(0), out var petShopId))
+				return CommandResult.InvalidArgument;
+
+			if (!ZoneServer.Instance.Data.CompanionDb.TryFind(petShopId, out var data))
+				return CommandResult.InvalidArgument;
+
+			if (!ZoneServer.Instance.Data.MonsterDb.TryFind(data.ClassName, out var monData))
+				return CommandResult.InvalidArgument;
+
+			var targetNpcName = "Companion Trader";
+			var currentNpcName = sender.Connection.CurrentDialog?.Npc.Name;
+
+			if (string.IsNullOrEmpty(currentNpcName) || !currentNpcName.Contains(targetNpcName))
+				return CommandResult.InvalidArgument;
+
+			//TODO: Decide if we sell pets that don't have a price defined.
+			if (data.Price == 0)
+				return CommandResult.Okay;
+
+			if (sender.Inventory.CountItem(ItemId.Silver) < data.Price)
+			{
+				sender.SystemMessage("OwnerDontHaveSilver");
+				return CommandResult.Okay;
+			}
+
+			if (sender.Inventory.Remove(ItemId.Silver, data.Price) != InventoryResult.Success)
+				return CommandResult.Fail;
+
+			var companion = new Companion(sender, monData.Id);
+			if (args.Count > 1)
+				companion.Name = args.Get(1);
+
+			sender.Companions.CreateCompanion(companion);
+
+			return CommandResult.Okay;
+		}
+
+		/// <summary>
+		/// Official slash command to raise pet stats
+		/// </summary>
+		/// <example>/petstat 528525790635969 MHP 1</example>
+		/// <param name="sender"></param>
+		/// <param name="target"></param>
+		/// <param name="message"></param>
+		/// <param name="command"></param>
+		/// <param name="args"></param>
+		/// <returns></returns>
+		private CommandResult HandlePetStat(Character sender, Character target, string message, string command, Arguments args)
+		{
+			// Since this command is sent via UI interactions, we'll not
+			// use any automated command result messages, but we'll leave
+			// debug messages for now, in case of unexpected values.
+			if (args.Count < 2)
+			{
+				Log.Debug("HandlePetStat: Invalid call by user '{0}': {1}", sender.Username, command);
+				return CommandResult.Okay;
+			}
+
+			if (!sender.Companions.HasCompanions)
+				return CommandResult.Okay;
+
+			if (long.TryParse(args.Get(0), out var companionObjectId))
+			{
+				var companion = sender.Companions.GetCompanion(companionObjectId);
+				var propertyName = "Monster_Stat_" + args.Get(1);
+
+				if (companion != null && PropertyTable.Exists("Monster", propertyName)
+					&& int.TryParse(args.Get(2), out var modifierValue))
+				{
+					var baseCost = propertyName == PropertyName.Stat_DEF ? 600 : 300;
+					var totalCost = 0;
+					var currentValue = companion.Properties.GetFloat(propertyName) - 1;
+
+					for (var i = currentValue; i < (currentValue + modifierValue); i++)
```

Comment:
This feels like it might be a good candidate for a scriptable function to allow for easy customization. What do you think?